src/nbsap/views/objectives.py

```diff
@@ -9,19 +9,44 @@
 from nbsap import models
 from nbsap.forms import NationalObjectiveForm, NationalObjectiveEditForm
 from nbsap.utils import remove_tags
+from nbsap.models import sort_by_code
 
 from auth import auth_required
 
 
-def nat_strategy(request, code=None):
+def get_adjacent_objectives(objectives, current_objective):
+    all_objectives = []
+    for objective in objectives:
+        all_objectives.append(objective)
+        all_objectives.extend(objective.get_all_objectives())
+    all_objectives = sort_by_code(all_objectives)
+    for index, objective in enumerate(all_objectives):
+        if objective == current_objective:
+            previous_index = index - 1
+            next_index = index + 1
+            break
+
+    if previous_index < 0:
+        previous_index = len(all_objectives) - 1
+    if next_index > len(all_objectives) - 1:
+        next_index = 0
+
+    return all_objectives[previous_index], all_objectives[next_index]
+
+
+def nat_strategy(request, pk=None):
     objectives = (
         models.NationalObjective.objects
         .filter(parent=None).order_by('id').all()
     )
     if not objectives.exists():
         return render(request, 'objectives/empty_nat_strategy.html')
-    code = code or objectives[0].code
-    current_objective = models.NationalObjective.objects.get(code=code)
+
+    for obj in objectives:
+        obj.objectives_tree = obj.get_all_objectives()
```

Comment:
The `objectives_tree` property is redundant. You should use `get_all_objectives` method. You can rename it to `objectives_tree` and also add the `@property` decorator to emulate a property behaviour.